### PR TITLE
13/add capability

### PR DESCRIPTION
**Pull request by Matthew-Steen**
## Description
Changes made for this branch

## Checklist
- [x] All manual tests passing
- [x] All unit tests passing 
- [x] All UI tests passing
- [x] Code reviewed by peer:\
&nbsp; - *No redundant whitespace*\
&nbsp; - *Indentation correct*\
&nbsp; - *Naming conventions followed*\
&nbsp; - *Comments added (where needed)*\
&nbsp; - *No commented out redundant code*\
&nbsp; - *Branch name correct*
- [x] Code reviewed by TA

### CODE REVIEW

File: Angular/src/app/add-capability/add-capability.component.html [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<div class="modal-header">
+    <h5 class="modal-title" id="exampleModalLabel">Add Capability</h5>
+    <button
+      type="button"
+      class="close"
+      aria-label="Close"
+      (click)="activeModal.dismiss()"
+    >
+      x
```
Comment (BenAAndrew) [inline Angular/src/app/add-capability/add-capability.component.html]: I would suggest going through and neatening up the HTML. Like putting this > on the line above it and removing empty lines

File: Angular/src/app/add-capability/add-capability.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  @Output() capabilityAdded = new EventEmitter();
+  newCapability: Capability;
+  public addCapabilityForm: FormGroup;
+  public submitted = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    public activeModal: NgbActiveModal,
+    private dataService: DataService,
+    private router: Router
```
Comment (BenAAndrew) [inline Angular/src/app/add-capability/add-capability.component.ts]: Do you use Router?

File: Angular/src/app/admin-table/admin-table.component.css [hunk 3]
```diff
@@ -47,4 +47,9 @@ table {
 .container {
   overflow-x: auto; 
   width: 90%;
+}
+
+#addCapability {
```
Comment (BenAAndrew) [inline Angular/src/app/admin-table/admin-table.component.css]: Remove this CSS class if all the contents are commented out

File: Angular/src/index.html [hunk 4]
```diff
@@ -15,6 +15,9 @@
   integrity="sha384-MCw98/SFnGE8fJT3GXwEOngsV7Zt27NXFoaoApmYm81iuXoPkFOJwJ8ERdknLPMO"
   crossorigin="anonymous"
   />
+
+  <link rel="stylesheet" 
+  href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
```
Comment (BenAAndrew) [inline Angular/src/index.html]: Is this needed?

File: Angular/src/app/add-capability/add-capability.component.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+          class="form-control"
+          [ngClass]="{ 'is-invalid': submitted && formControls.capabilityName.errors }"
+        />
+        <div
+          *ngIf="submitted && formControls.capabilityName.errors"
+          class="invalid-feedback"
+        >
+          <div *ngIf="formControls.capabilityName.errors.required">
+            Capability Name is a required field
+          </div>
```
Comment (matthewdoreilly) [inline Angular/src/app/add-capability/add-capability.component.html]: Just always watch the indentation/formatting of your HTML. It seems minor, but it's important for readability. This `div` should be decremented back one

File: Angular/src/app/add-capability/add-capability.component.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    this.newCapability = {
+      departmentId: this.departmentId,
+      name: this.addCapabilityForm.get("capabilityName").value
+    }
+  }
+
+  onSubmit() {
+    this.submitted = true;
+    if (this.addCapabilityForm.invalid) {
+      return;
```
[the remark this answers is not in the document]
Comment (Matthew-Steen) [reply Angular/src/app/add-capability/add-capability.component.ts]: Errors are shown in the HTML, yes.

File: Angular/src/app/model/capability.ts [hunk 7]
```diff
@@ -0,0 +1,4 @@
+export class Capability {
```
Comment (matthewdoreilly) [inline Angular/src/app/model/capability.ts]: I think you can just make this an `interface`:

```
export interface Capability {
     departmentId: number;
     name: string;
}
```

See https://stackoverflow.com/a/51716899

File: Angular/src/app/admin-table/admin-table.component.css [hunk 8]
```diff
@@ -47,4 +47,9 @@ table {
...
+  /* padding: 10px 10px; */
+  /* margin: 5px 5px; */
```
[the remark this answers is not in the document]
Comment (Matthew-Steen) [reply Angular/src/app/admin-table/admin-table.component.css]: Done

**Pull request merged**